CHANGELOG.md

```diff
@@ -4,9 +4,77 @@ This document includes a curated changelog for each release. We also publish a c
 a [GitHub release](https://github.com/nginxinc/nginx-gateway-fabric/releases), which, by contrast, is auto-generated
 and includes links to all PRs that went into the release.
 
+## Release 1.3.0
+
+_June 11, 2024_
+
+FEATURES:
+
+- The minimum supported version of Kubernetes is now 1.25. [1885](https://github.com/nginxinc/nginx-gateway-fabric/pull/1885)
```

Comment:
I noticed that in previous changelogs we used to have it like `PR-1885`. I don't think we need to add `PR-` and I prefer it without, but just pointing it out...